pallets/payment/src/types.rs

```diff
@@ -77,7 +86,7 @@ pub trait DisputeResolver<Account> {
 }
 
 /// Fee Handler trait that defines how to handle marketplace fees to every payment/swap
-pub trait FeeHandler<Account> {
+pub trait FeeHandler<Account, Remark> {
 	/// Get the distribution of fees to marketplace participants
-	fn apply_fees(from: &Account, to: &Account) -> (Account, Percent);
+	fn apply_fees(from: &Account, to: &Account, remark: &Option<Remark>) -> (Account, Percent);
```

Comment:
It won't be very useful to users of this pallet that implement their own version of the trait to have the remark as a generic type without bounds of any kind so they can use the data somehow, but what bounds should it have? Instead of trying to restrict the remark to follow some specific behavior/trait I'd say we just use the type as is(e.g. `Vec<u8>`) or more useful for developers I think it would be the `PaymentDetail` to have more data to make different decisions. Since we construct the type after we call the `apply_fees` one option could be to make that `fee_detail` property an `Option` to be able to pass the detail to the function. Another option that favors more the encapsulation principle is to have a separate type that is a redacted version of the `PaymentDetail` and has a subset of the fields and then it can be converted to the payment detail.

e.g.
```rust
struct PaymentInfo<Asset, Amount> {
        pub asset: Asset,
	pub amount: Amount,
        pub remark: Vec<u8>,
}

// then some means to convert
impl From<PaymentInfo> for PaymentDetail { ... }
```
... or something like that